### DISCUSSION TITLE
Can we merge multiple pdfs into single pdf file using Imagemagick?

### DISCUSSION THREAD

**Discussion by tusharg304** [ImageMagick/ImageMagick#3559] [Help]
Hello Everyone,

I am new to imagemagick and need to know if we can merge multiple pdfs into single pdf file using Imagemagick?
Appreciate your help in adance!

Thanks

**Comment by snibgo** [comment]
Yes, you can. For example, you can make one output pdf from two input pdfs:
```
magick in1.pdf in2.pdf out.pdf
```
However, ImageMagick is a raster image processor. The command will convert each input page to a raster (pixel) image, and write these raster images to the output. If your inputs contain selectable text, no selectable text will be present in the output.

The common utility program "pdfunite" may be more suitable for you.

**Reply by tusharg304** [reply]
I tried above command but I am getting an error : "magick: FailedToExecuteCommand `"gswin64c.exe""
Is this exe file supposed to be part of imagemagick?

**Reply by fmw42** [reply]
Do you have Ghostscript installed with Imagemagick? It is needed for reading PDF files.  Do you get that error if you just read a single pdf file and save as gif? If so, you likely do not have Ghostscript installed.

**Reply by snibgo** [reply]
gswin64c.exe is Ghostscript, which isn't distributed with ImageMagick. IM uses GS to rasterize PDFs. GS can be downloaded from https://ghostscript.com/

**Reply by tusharg304** [reply]
Thank You!
Installed ghostscript and now able to merge the pdfs, but resolution and clarity of merged file is not good and some of the content is not even readable. Original PDF files are very much clear and readable as compared to merged PDF.
Are there any other tools that I can use? I need to merge like 60-70 pdf files into one.

**Reply by snibgo** [reply]
I have suggested: "The common utility program "pdfunite" may be more suitable for you."

Jmuccigr has suggested "I'm a fan of qpdf".

**Reply by tusharg304** [reply]
I was able to get it going on windows 10. 
Able to merge the PDF files with clear resolution as well using below command
magick convert -density 200 "File1.PDF" "File2.PDF" "Output.PDF"

But I can't seem to get it working on Windows Server 2019. Output PDF is generated but i can not open it. Error: File is damaged and can not be repaired.
Is Imagemagick supposed to be working with Windows server 2019. Do you have any specific version recommendation for this?

Thanks,
TG

**Reply by Jmuccigr** [reply]
Again, IM is not the right tool for this. The other things we have mentioned will merge your PDFs without changing the originals.

**Reply by snibgo** [reply]
> Is Imagemagick supposed to be working with Windows server 2019.

Yes. What versions of IM and Ghostscript do you use on the two systems?

**Reply by tusharg304** [reply]
ImageMagick-7.0.11-8-Q16-HDRI-x64-dll.exe

Ghostscript 9.53.3 for Windows (64 bit) | Ghostscript AGPL Release

**Reply by snibgo** [reply]
The same versions? Then I don't know why it works on one but not the other.

**Reply by fmw42** [reply]
Perhaps running out of resources or temp space?

**Comment by Jmuccigr** [comment]
As @snibgo suggests, other tools are better for this. I'm a fan of qpdf, but there are others, as he notes.

**Reply by tusharg304** [reply]
@Jmuccigr Can you please provide details about qpdf?

**Reply by Jmuccigr** [reply]
http://qpdf.sourceforge.net/
